### PR TITLE
Account Gateway, SimpleAPI, Legacy Service, TotemDB, Demo scene

### PR DESCRIPTION
**Pull request by voropserg**

### CODE REVIEW

File: services/TotemAccountGateway.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using UnityEngine.Events;
+using System.Net;
+using utilities;
+
+
+namespace TotemServices
+{
+    public class TotemAccountGateway : MonoBehaviour
+    {
+        private readonly string _urlRoot = "https://account.totem.gdn/";
```
Comment (snirsh) [inline services/TotemAccountGateway.cs]: Do you thing those should be in an .env or similar config file instead?

File: services/TotemAccountGateway.cs [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+        private readonly string _httpListenerUrl = "http://localhost:6700/auth/";
+
+        private readonly string _succesResponse = "<html><head><meta charset='utf8'></head><body>Login successful. You can close this page and return to the app</body></html>";
```
Comment (snirsh) [inline services/TotemAccountGateway.cs]: Responses file maybe?

File: services/TotemAccountGateway.cs [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        }
+
+        private IEnumerator GetPublicKeyCoroutine(SocialLoginResponse loginInfo, UnityAction<string, UserSocialProfile> onSuccess, UnityAction<string> onFailure)
+        {
+            UnityWebRequest www = UnityWebRequest.Get(_urlRoot + "me");
+            www.SetRequestHeader("Authorization", "Bearer " + loginInfo.accessToken);
+            yield return www.SendWebRequest();
+            if (www.result == UnityWebRequest.Result.ConnectionError || www.result == UnityWebRequest.Result.ProtocolError)
+            {
+                Debug.LogError("TotemAccountGateway- Failed to get public key: " + www.error);
```
Comment (snirsh) [inline services/TotemAccountGateway.cs:100]: Again this can be in a responses consts file

File: services/TotemLegacyService.cs [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+using UnityEngine.Events;
+using UnityEngine.Networking;
+using TotemEntities;
+
+
+namespace TotemServices
+{
+    public class TotemLegacyService : MonoBehaviour
+    {
+        private readonly string _rootUrl = "https://legacy-api.totem.gdn/";
```
Comment (snirsh) [inline services/TotemLegacyService.cs]: Same as before, maybe .env file and a configuration service

File: services/TotemSimpleAPI.cs [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            StartCoroutine(GetItemsCoroutine(publicKey, onSuccess, onFailure));
+        }
+
+        private IEnumerator GetItemsCoroutine(string publicKey, UnityAction<List<TotemSpear>> onSuccess, UnityAction<string> onFailure = null)
+        {
+            UnityWebRequest www = UnityWebRequest.Get(_itemsUrl + publicKey);
+            yield return www.SendWebRequest();
+            if (www.result == UnityWebRequest.Result.ConnectionError || www.result == UnityWebRequest.Result.ProtocolError)
+            {
+                Debug.LogError("TotemSimpleAPI- Failed to get items: " + www.error);
```
Comment (snirsh) [inline services/TotemSimpleAPI.cs:76]: Export to const file

File: services/TotemSimpleAPI.cs [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+            StartCoroutine(GetAvatasCoroutine(publicKey, onSuccess, onFailure));
+        }
+
+        private IEnumerator GetAvatasCoroutine(string publicKey, UnityAction<List<TotemAvatar>> onSuccess, UnityAction<string> onFailure = null)
+        {
+            UnityWebRequest www = UnityWebRequest.Get(_avatarsUrl + publicKey);
+            yield return www.SendWebRequest();
+            if (www.result == UnityWebRequest.Result.ConnectionError || www.result == UnityWebRequest.Result.ProtocolError)
+            {
+                Debug.LogError("TotemSimpleAPI- Failed to get avatars: " + www.error);
```
Comment (snirsh) [inline services/TotemSimpleAPI.cs:108]: Const file

**Review by snirsh — CHANGES_REQUESTED** [review]
Thinking about creating an .env configurations file and a HTTPRespinses consts file that will contain the messages. What do you think?

File: services/TotemAccountGateway.cs (hunk 1 above)
Comment (voropserg) [reply services/TotemAccountGateway.cs]: Yes, thats a good idea to keep this string-links in an env const file.

File: services/TotemAccountGateway.cs (hunk 2 above)
Comment (voropserg) [reply services/TotemAccountGateway.cs]: Yea, i'll create a HTML file for the response

**Comment by voropserg** [comment]
I've moved all the links to the services.env file. Made a .html response file for the socials login (for possible future improvement). But decided to keep the error messages in the coresponding files, since it's bascly jsut a flawor text and the error it self is in the www.error

**Review by snirsh — APPROVED** [review]
🔥

**Pull request merged**